docs(destination-clickhouse): Improve documentation with context and present tense (#69223)
airbytehq/airbyte: docs/integrations/destinations/clickhouse-migrations.md

### Migrating Existing Data to the New Format

Airbyte cannot automatically migrate data from the v1 raw table format to the v2 typed table format. To get your data into the new format, you must perform a full refresh sync from your source.

**Migration steps:**

1. Upgrade your ClickHouse destination connector to version 2.0.0 or later
2. Trigger a full refresh sync to populate the new typed tables
3. Verify the new tables contain the expected data
4. Update your downstream pipelines to reference the new table locations
5. Optional: Remove the old v1 raw tables (see below)

#### Optional: Removing the Old v1 Raw Tables

The v2 connector does not automatically remove tables created by the v1 connector. After successfully migrating to v2 and verifying your data, you can optionally remove the old raw tables created by v1 to free up storage space.

For most users, old tables are located in the `airbyte_internal` database with names matching the pattern `airbyte_internal.{database}_raw__stream_{table}`. However, the exact location may vary based on your v1 configuration.

:::caution

> Before removing any tables, pause all Airbyte connections that write to this ClickHouse instance and verify that no syncs are running. The v2 ClickHouse destination uses the `airbyte_internal` database for temporary scratch space (for example, streams running in `dedup` mode, truncate refreshes, and overwrite syncs). Dropping the entire `airbyte_internal` database can interrupt active syncs and cause data loss. Only drop the specific v1 raw tables you no longer need.

Reviewer: ```suggestion
The v2 ClickHouse destination uses the `airbyte_internal` database for temporary scratch space (for example, streams running in `dedup` mode, truncate refreshes, and overwrite syncs). Dropping the entire `airbyte_internal` database can interrupt active syncs and cause data loss. Only drop the specific v1 raw tables you no longer need.
```

# ClickHouse Migration Guide

## SSH Support

:::warning
SSH tunneling support for the ClickHouse connector is currently in **Beta**.
:::

## Upgrading to 2.0.0

Version 2.0.0 represents a fundamental architectural change from version 1.0.0.

**Version 1.0.0 behavior:**

- Wrote all data as JSON to raw tables in the `airbyte_internal` database
- Used table names like `airbyte_internal.{database}_raw__stream_{table}`

**Version 2.0.0 behavior:**

- Writes data to typed columns matching your source schema
- Creates tables in the configured database with clean table names: `{database}.{table}`
- No longer uses the `airbyte_internal` database or `_raw__stream_` prefixes

While this is a breaking change, existing connections continue to function after upgrading. However, data is written to a completely different location in a different format. **You must update any downstream pipelines** (SQL queries, BI dashboards, data transformations) to reference the new table locations and schema structure.

> 

Reviewer 2: I do _not_ like removing the gotchas header. This draws attention to very specific issues encountered by a high number of users and how to fix them. The new format makes these less obvious by nesting them in a broader categorical information architecture.

# ClickHouse Migration Guide

## SSH Support

:::warning
SSH tunneling support for the ClickHouse connector is currently in **Beta**. If you encounter issues with SSH tunneling after upgrading, please contact Airbyte support.
:::

## Upgrading to 2.0.0

> Version 2.0.0 represents a fundamental architectural change from version 1.0.0:

Reviewer 3: Finish this with a period instead of a colon.

**Version 1.0.0 behavior:**

- Wrote all data as JSON to raw tables in the `airbyte_internal` database
- Used table names like `airbyte_internal.{database}_raw__stream_{table}`

**Version 2.0.0 behavior:**

- Writes data to typed columns matching your source schema
- Creates tables in the configured database with clean table names: `{database}.{table}`
- No longer uses the `airbyte_internal` database or `_raw__stream_` prefixes

While this is a breaking change, existing connections continue to function after upgrading. However, data is written to a completely different location in a different format. **You must update any downstream pipelines** (SQL queries, BI dashboards, data transformations) to reference the new table locations and schema structure.

## Migrating Existing Data to the New Format

Airbyte cannot automatically migrate data from the v1 raw table format to the v2 typed table format. To get your data into the new format, you must perform a full refresh sync from your source.

**Migration steps:**

1. Upgrade your ClickHouse destination connector to version 2.0.0 or later

> 2. Update your downstream pipelines to reference the new table locations

Reviewer 3: Wouldn't steps 3-4 need to happen before step 2? Otherwise dependencies will be referencing nothing. Also isn't step 5 just a repeat of step 2?

3. Verify the new tables contain the expected data
4. Update your downstream pipelines to reference the new table locations
5. Optional: Remove the old v1 raw tables (see below)

#### Optional: Removing the Old v1 Raw Tables

The v2 connector does not automatically remove tables created by the v1 connector. After successfully migrating to v2 and verifying your data, you can optionally remove the old raw tables created by v1 to free up storage space.

For most users, old tables are located in the `airbyte_internal` database with names matching the pattern `airbyte_internal.{database}_raw__stream_{table}`. However, the exact location may vary based on your v1 configuration.

:::caution
Always verify that you have successfully migrated your data before removing old tables. Once dropped, the data cannot be recovered without re-syncing from your source.
:::

To remove old tables:

```sql
-- List tables in the airbyte_internal database
SHOW TABLES FROM airbyte_internal;

-- Drop individual tables
DROP TABLE airbyte_internal.{database}_raw__stream_{table};

-- Or drop the entire airbyte_internal database if it only contains old Airbyte tables

> DROP DATABASE airbyte_internal;

Reviewer: Users should verify no syncs are running before they do this. The current connector does still use the `airbyte_internal` database for temporary scratch space (dedup streams will stage data into this database, and truncate refreshes / overwrite syncs will store data here as well).

For most users, old tables are located in the `airbyte_internal` database with names matching the pattern `airbyte_internal.{database}_raw__stream_{table}`. However, the exact location may vary based on your v1 configuration.

:::caution

> Before removing any tables, pause all Airbyte connections that write to this ClickHouse instance and verify that no syncs are running. The v2 ClickHouse destination uses the `airbyte_internal` database for temporary scratch space (for example, deduplication streams, truncate refreshes, and overwrite syncs). Dropping the entire `airbyte_internal` database can interrupt active syncs and cause data loss. Only drop the specific v1 raw tables you no longer need.

Reviewer: ```suggestion
The v2 ClickHouse destination uses the `airbyte_internal` database for temporary scratch space (for example, streams running in `dedup` mode, truncate refreshes, and overwrite syncs). Dropping the entire `airbyte_internal` database can interrupt active syncs and cause data loss. Only drop the specific v1 raw tables you no longer need.
```
